src/utils/ngd/url.ts

```diff
@@ -0,0 +1,57 @@
+// src/utils/ngd/url.ts
+import { BBox } from "../../types";
+import { getCRS } from "../crs";
+
+export { buildUrl };
+
+function buildUrl(
+  collectionId: string,
+  {
+    featureId = null,
+    bbox = null,
+    datetime = null,
+    filter = null,
+    crs = null,
+    bboxCRS = null,
+    filterCRS = null,
+  }: {
+    featureId?: string | null;
+    bbox?: null | BBox;
+    datetime?: null | string;
+    filter?: null | string;
+    crs?: null | number | string;
+    bboxCRS?: null | number | string;
+    filterCRS?: null | number | string;
+  } = {}
+) {
+  const root = "https://api.os.uk/features/ngd/ofa/v1/collections/";
+  let queryParams: {
+    bbox?: string;
+    datetime?: string;
+    filter?: string;
+    crs?: string;
+    "bbox-crs"?: string;
+    "filter-crs"?: string;
+  } = {};
+  if (bbox) {
+    queryParams["bbox"] = bbox.join(",");
```

Comment:
Should we be doing some basic checks (such as minx < maxx)

Reply:
Added validation